src/graphql/data/nft/Asset.ts

```diff
@@ -0,0 +1,176 @@
+import { AssetQuery, NftAssetsFilterInput, NftAssetSortableField } from './__generated__/AssetQuery.graphql'
+
+const assetPaginationQuery = graphql`
+  fragment AssetQuery_nftAssets on Query @refetchable(queryName: "AssetPaginationQuery") {
+    nftAssets(
+      address: $address
+      orderBy: $orderBy
+      asc: $asc
+      filter: $filter
+      first: $first
+      after: $after
+      last: $last
+      before: $before
+    ) @connection(key: "AssetQuery_nftAssets") {
+      edges {
+        node {
+          id
+          name
+          ownerAddress
+          image {
+            url
+          }
+          smallImage {
+            url
+          }
+          originalImage {
+            url
+          }
+          tokenId
+          description
+          animationUrl
+          suspiciousFlag
+          collection {
+            name
+            isVerified
+            image {
+              url
+            }
+            creator {
+              address
+              profileImage {
+                url
+              }
+              isVerified
+            }
+            nftContracts {
+              address
+              standard
+            }
+          }
+          listings(first: 1) {
+            edges {
+              node {
+                address
+                createdAt
+                endAt
+                id
+                maker
+                marketplace
+                marketplaceUrl
+                orderHash
+                price {
+                  currency
+                  value
+                }
+                quantity
+                startAt
+                status
+                taker
+                tokenId
+                type
+              }
+              cursor
+            }
+          }
+          rarities {
+            provider
+            rank
+            score
+          }
+          metadataUrl
+        }
+      }
+    }
+  }
+`
+
+const assetQuery = graphql`
+  query AssetQuery(
+    $address: String!
+    $orderBy: NftAssetSortableField
+    $asc: Boolean
+    $filter: NftAssetsFilterInput
+    $first: Int
+    $after: String
+    $last: Int
+    $before: String
+  ) {
+    ...AssetQuery_nftAssets
+  }
+`
+
+export function useAssetsQuery(
+  address: string,
+  orderBy: NftAssetSortableField,
+  asc: boolean,
+  filter: NftAssetsFilterInput,
+  first?: number,
+  after?: string,
+  last?: number,
+  before?: string
+) {
+  const assetsQueryReference = loadQuery<AssetQuery>(RelayEnvironment, assetQuery, {
+    address,
+    orderBy,
+    asc,
+    filter,
+    first,
+    after,
+    last,
+    before,
+  })
+  const queryData = usePreloadedQuery<AssetQuery>(assetQuery, assetsQueryReference)
+  const { data, hasNext, loadNext, isLoadingNext } = usePaginationFragment<AssetPaginationQuery, any>(
+    assetPaginationQuery,
+    queryData
+  )
+
+  const assets: GenieAsset[] = data.nftAssets?.edges?.map((queryAsset: { node: any }) => {
```

Comment:
is the intention to eventually match the graphql schema so you don't have to manipulate it?

Reply:
The intention was to have an intermediary function that sanitizes the input into the type of asset we use on the front end. A later on tech debt could be a major refactor of the GenieAsset type to match GQL but that is out of scope for this migration

Reply:
yeah definitely not suggesting you do it before phase 1, was just curious if that would be an eventual goal